examples/energy-management-app/energy-management-common/device-energy-management/src/device-energy-management-mode.cpp

```diff
@@ -110,7 +110,26 @@ void emberAfDeviceEnergyManagementModeClusterInitCallback(chip::EndpointId endpo
     gDeviceEnergyManagementModeInstance->Init();
 }
 
-void MatterDeviceEnergyManagementModeClusterServerShutdownCallback(chip::EndpointId endpoint)
+void emberAfDeviceEnergyManagementModeClusterShutdownCallback(chip::EndpointId endpointId)
 {
+    if (endpointId != GetEnergyDeviceEndpointId())
+    {
+        return;
+    }
+
+    if (gDeviceEnergyManagementModeInstance)
+    {
+        gDeviceEnergyManagementModeInstance->Shutdown();
+    }
+    DeviceEnergyManagementMode::Shutdown();
+}
+
+void MatterDeviceEnergyManagementModeClusterServerShutdownCallback(chip::EndpointId endpointId)
```

Comment:
Is this ember generated callback? who are going to call this function?

Reply:
Yes it is a generated callback. `Server::Shutdown()` will call it PTAL https://github.com/project-chip/connectedhomeip/pull/38515